### PR TITLE
Update to tensorflow 1.0.1

### PR DESCRIPTION
**Pull request by javyxx**
Update to tensorflow 1.0.1

### CODE REVIEW

File: src/tinyYolov2/TinyYolov2.ts [hunk 1]
```diff
@@ -205,12 +205,13 @@ export class TinyYolov2 extends NeuralNetwork<TinyYolov2NetParams> {
...
+          const score = sigmoid(scoresTensor.arraySync()[row][col][anchor][0]);
           if (!scoreThreshold || score > scoreThreshold) {
-            const ctX = ((col + sigmoid(boxesTensor.get(row, col, anchor, 0))) / numCells) * correctionFactorX
-            const ctY = ((row + sigmoid(boxesTensor.get(row, col, anchor, 1))) / numCells) * correctionFactorY
-            const width = ((Math.exp(boxesTensor.get(row, col, anchor, 2)) * this.config.anchors[anchor].x) / numCells) * correctionFactorX
-            const height = ((Math.exp(boxesTensor.get(row, col, anchor, 3)) * this.config.anchors[anchor].y) / numCells) * correctionFactorY
+            const ctX = ((col + sigmoid(boxesTensor.arraySync()[row][col][anchor][0])) / numCells) * correctionFactorX
+            const ctY = ((row + sigmoid(boxesTensor.arraySync()[row][col][anchor][1])) / numCells) * correctionFactorY
+            const width = ((Math.exp(boxesTensor.arraySync()[row][col][anchor][2]) * this.config.anchors[anchor].x) / numCells) * correctionFactorX
+            const height = ((Math.exp(boxesTensor.arraySync()[row][col][anchor][3]) * this.config.anchors[anchor].y) / numCells) * correctionFactorY
```
Comment (justadudewhohacks) [inline src/tinyYolov2/TinyYolov2.ts]: I am not sure if it's a good idea to call arraySync() on every iteration. I'd rather get the data once outside of the loop:

``` javascript 
  const scoresData = await scoresTensor.array()
  const boxesData = await boxesTensor.array()
```

Also you can turn this function into an async one.

File: src/tinyYolov2/TinyYolov2.ts [hunk 2]
```diff
@@ -242,7 +243,7 @@ export class TinyYolov2 extends NeuralNetwork<TinyYolov2NetParams> {
   private extractPredictedClass(classesTensor: tf.Tensor4D, pos: { row: number, col: number, anchor: number }) {
     const { row, col, anchor } = pos
```
Comment (justadudewhohacks) [inline src/tinyYolov2/TinyYolov2.ts]: Same here: `const classesData = await classesTensor.array()` and make this function async.

File: src/tinyYolov2/TinyYolov2.ts [hunk 3]
```diff
@@ -205,12 +205,13 @@ export class TinyYolov2 extends NeuralNetwork<TinyYolov2NetParams> {
...
+          const score = sigmoid(scoresTensor.arraySync()[row][col][anchor][0]);
           if (!scoreThreshold || score > scoreThreshold) {
-            const ctX = ((col + sigmoid(boxesTensor.get(row, col, anchor, 0))) / numCells) * correctionFactorX
-            const ctY = ((row + sigmoid(boxesTensor.get(row, col, anchor, 1))) / numCells) * correctionFactorY
-            const width = ((Math.exp(boxesTensor.get(row, col, anchor, 2)) * this.config.anchors[anchor].x) / numCells) * correctionFactorX
-            const height = ((Math.exp(boxesTensor.get(row, col, anchor, 3)) * this.config.anchors[anchor].y) / numCells) * correctionFactorY
+            const ctX = ((col + sigmoid(boxesTensor.arraySync()[row][col][anchor][0])) / numCells) * correctionFactorX
+            const ctY = ((row + sigmoid(boxesTensor.arraySync()[row][col][anchor][1])) / numCells) * correctionFactorY
+            const width = ((Math.exp(boxesTensor.arraySync()[row][col][anchor][2]) * this.config.anchors[anchor].x) / numCells) * correctionFactorX
+            const height = ((Math.exp(boxesTensor.arraySync()[row][col][anchor][3]) * this.config.anchors[anchor].y) / numCells) * correctionFactorY
```
Comment (justadudewhohacks) [inline src/tinyYolov2/TinyYolov2.ts]:
``` javascript
const ctX = ((col + sigmoid(boxesData[row][col][anchor][0])) / numCells) * correctionFactorX
const ctY = ((row + sigmoid(boxesData[row][col][anchor][1])) / numCells) * correctionFactorY
const width = ((Math.exp(boxesData[row][col][anchor][2]) * this.config.anchors[anchor].x) / numCells) * correctionFactorX
const height = ((Math.exp(boxesData[row][col][anchor][3]) * this.config.anchors[anchor].y) / numCells) * correctionFactorY
```

File: src/tinyYolov2/TinyYolov2.ts [hunk 4]
```diff
@@ -205,12 +205,13 @@ export class TinyYolov2 extends NeuralNetwork<TinyYolov2NetParams> {
     for (let row = 0; row < numCells; row ++) {
       for (let col = 0; col < numCells; col ++) {
         for (let anchor = 0; anchor < numBoxes; anchor ++) {
-          const score = sigmoid(scoresTensor.get(row, col, anchor, 0))
+
+          const score = sigmoid(scoresTensor.arraySync()[row][col][anchor][0]);
```
Comment (justadudewhohacks) [inline src/tinyYolov2/TinyYolov2.ts]:
``` javascript
const score = sigmoid(scoresData[row][col][anchor][0])
```

**Comment by justadudewhohacks** [comment]
Thanks for your PR! I added some minor remarks.

File: src/tinyYolov2/TinyYolov2.ts (hunk 1 above)
Comment (javyxx) [reply src/tinyYolov2/TinyYolov2.ts]: If you're right. I focused on direct migration and I did not realize those optimizations

**Comment by javyxx** [comment]
> Thanks for your PR! I added some minor remarks.

They are not minor comments. These are very important details for the performance.

**Review by javyxx — COMMENTED** [review]
I have applied the suggested changes

**Comment by justadudewhohacks** [comment]
Awesome thanks! One last request. Please remove everything from build/ from your pull request. I will check in the build manually after merging.

**Comment by javyxx** [comment]
Done

**Comment by justadudewhohacks** [comment]
Sorry if I wasn't clear, I wanted you to remove the build from the PR by reverting the changes, not to delete it.

**Pull request closed without merging**